docs/posts/1tbc/index.qmd

```diff
@@ -0,0 +1,755 @@
+---
+title: "Querying 1TB on a laptop with Python dataframes"
+author: "Cody Peterson"
+date: "2024-07-08"
+image: ibis-duckdb-sort.gif
+categories:
+    - benchmark
+    - duckdb
+    - datafusion
+    - polars
+---
+
+***TPC-H benchmark at `sf=1024` via DuckDB, DataFusion, and Polars on a MacBook
+Pro with 96GiB of RAM.***
+
+---
+
+pandas requires your dataframe to fit in memory. Out-of-memory (OOM) errors are
+common when working on larger datasets, though the corresponding size of data on
+disk can be surprising. The creator of pandas and Ibis noted in ["Apache
+Arrow and the '10 Things I Hate About
+pandas'"](https://wesmckinney.com/blog/apache-arrow-pandas-internals):
+
+> To put it simply, **we weren’t thinking about analyzing 100 GB or 1 TB datasets
+> in 2011**. [In 2017], my rule of thumb for pandas is that **you should have 5 to
+> 10 times as much RAM as the size of your dataset**. So if you have a 10 GB
+> dataset, you should really have about 64, preferably 128 GB of RAM if you want
+> to avoid memory management problems. This comes as a shock to users who expect
+> to be able to analyze datasets that are within a factor of 2 or 3 the size of
+> their computer’s RAM.
+
+Today with Ibis you can reliably and efficiently process a 1TB dataset on a
+laptop with <1/10th the RAM.
```

Comment:
Perhaps worth stating this as 100x **LESS** RAM than recommended by Wes as opposed to 1/10th?